File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        longestStr = ""
        for i in range(len(strs[0])):
            for s in strs:
                if i == len(s) or s[i] != strs[0][i]:
                    return longestStr
            longestStr += strs[0][i]

        return longestStr

        

File: Data Structures & Algorithms/anagram-groups/submission-0.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let res = {}
        for(let str of strs){
            let sstr = str.split("").sort().join()
            if(!res[sstr]){
                res[sstr] = []
            }
            res[sstr].push(str)  
        }  
        return Object.values(res);
    }
}
